src/abe_policy/access_policy.rs

```diff
@@ -360,13 +360,15 @@ impl AccessPolicy {
                     .get(&attr.axis)
                     .ok_or_else(|| Error::AxisNotFound(attr.axis.to_string()))?;
                 let mut res = vec![vec![attr.clone()]];
-                if axis_parameters.is_hierarchical && follow_hierarchical_axes {
-                    // add attribute values for all attributes below the given one
-                    for name in &axis_parameters.attribute_names {
-                        if *name == attr.name {
-                            break;
+                if follow_hierarchical_axes {
```

Comment:
We should be able to find a better name for this parameter... maybe `include_lower_attributes` or something like that. This would avoid the need for your comment below by making explicit in which direction hierarchical axes are followed.

Reply:
This check could be performed only if the axis is hierarchical btw (by switching this line with the following `if let Some` test).